config/src/main/java/com/yahoo/config/subscription/impl/GenericJRTConfigSubscription.java

```diff
@@ -36,7 +36,9 @@ public GenericJRTConfigSubscription(ConfigKey<?> key,
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     protected void setNewConfig(JRTClientConfigRequest jrtReq) {
+        setConfig(jrtReq.getNewGeneration(), null );
```

Comment:
this probably also has its own race-condition; signalling a new config is available before it actually is...